homeassistant/components/axis/binary_sensor.py

```diff
@@ -51,26 +51,21 @@ async def async_setup_entry(
     device: AxisNetworkDevice = hass.data[AXIS_DOMAIN][config_entry.unique_id]
 
     @callback
-    def async_add_sensor(event_id):
-        """Add binary sensor from Axis device."""
-        event: AxisEvent = device.api.event[event_id]
-
-        if event.group not in (CLASS_OUTPUT, CLASS_PTZ) and not (
-            event.group == CLASS_LIGHT and event.type == "Light"
-        ):
-            async_add_entities([AxisBinarySensor(event, device)])
-
-    config_entry.async_on_unload(
-        async_dispatcher_connect(hass, device.signal_new_event, async_add_sensor)
+    def async_create_entity(event: Event) -> None:
+        """Create Axis binary sensor entity."""
+        async_add_entities([AxisBinarySensor(event, device)])
+
+    device.api.event.subscribe(
+        async_create_entity,
+        topic_filter=EVENT_TOPICS,
```

Comment:
Would it make sense to make `EVENT_TOPICS` a `set` to avoid linear searching?

Reply:
I'm not sure, as there will ever only be a one to many check as an event can only have one topic so I don't think its relevant